src/database/preparedQueries.js

```diff
@@ -23,6 +23,7 @@ async function formatInsertArray(req, client) {
   }
 }
 
+//several issues with this method, more details to come on github comment
 async function addingIfNoID(req, client) {
```

Comment:
this kind of manual manipulation of the ids is not recommended, you need to split the data up before sending it to the server and only query the relevant endpoints

if you want to use a single POST endpoint like you are you can split the object into three sections for modify/insert/delete like

`{
    modify: [task, task task].
    insert: [task, task, task],
   delete: [task, task, task]
}`

Reply:
to be clear, I recommend setting up the different HTTP method endpoints, `PATCH`, `DELETE`

you also need to consider how you handle updating on the UI, one option is to send the resolved(final) list back after each operation and re-render

a common pattern is to submit the change and if its OK (HTTP status 200, 201) then do a partial re-render of the list. This can be a full re-render in current project for ease.